File: Java/ToyLanguageGUI_exam_final/src/domain/expressions/ArithmeticExpression.java
package domain.expressions;

import exceptions.DivisionException;
import utils.IDictionary;
import utils.IHeap;

public class ArithmeticExpression implements Expression {
    private  char operator;
    private Expression left,right;

    public ArithmeticExpression(char op,Expression left,Expression right){
        this.operator = op;
        this.left = left;
        this.right = right;
    }

    @Override
    public int eval(IDictionary<String,Integer> symTable, IHeap<Integer,Integer> heap) {
        int left1 = left.eval(symTable,heap);
        int right1 = right.eval(symTable,heap);
        if (operator == '+')
            return left1 + right1;
        if (operator == '-')
            return left1 - right1;
        if (operator == '*')
            return left1 * right1;
        if (operator == '/')
            if (right1 == 0)
                throw new DivisionException("Division by zero");
            return left1 / right1;
    }

    @Override
    public String toString(){
        return left.toString()+operator+right.toString();
    }
}

File: ToyInterpretor/ToyInterpretor/Domain/Expressions/ArithmeticExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Utils;
using ToyInterpretor.Exceptions;

namespace ToyInterpretor.Domain.Expressions
{
    class ArithmeticExpression : Expression
    {
        private Char op;
        private Expression left, right;

        public ArithmeticExpression(char op, Expression left, Expression right)
        {
            this.op = op;
            this.left = left;
            this.right = right;
        }

        public int eval(ISymTable<String, int> symTable)
        {
            int left1 = left.eval(symTable);
            int right1 = right.eval(symTable);
            if (op == '+')
            return left1 + right1;
            if (op == '-')
            return left1 - right1;
            if (op == '*')
            return left1 * right1;
            if (op == '/')
            if (right1 == 0)
                throw new DivisionException("Division by zero");
            return left1 / right1;
        }

        public override String ToString()
        {
            return left.ToString() +op+right.ToString();
        }
    }
}
